.github/workflows/CI.yml

```diff
@@ -116,3 +116,54 @@ jobs:
       - name: Test
         run: |
           ${{ matrix.config.qemu }} build/uv_run_tests_a
+
+  build-cross-mingw:
+    runs-on: ubuntu-latest
+    name: build-cross-mingw
+    steps:
+      - uses: actions/checkout@v2
+      - name: Envinfo
+        run: npx envinfo
+      - name: Install mingw32 environment
+        run: |
+          sudo apt update
+          sudo apt install mingw-w64 ninja-build -y
+      - name: Build
+        run: |
+          mkdir build
+          cd build && cmake .. -G Ninja -DBUILD_TESTING=ON -DCMAKE_TOOLCHAIN_FILE=../cmake-toolchains/cross-mingw32-linux.cmake
+          cmake --build .
+          ls -lh
+      - name: Upload build artifacts
+        uses: actions/upload-artifact@v2
+        with:
+          name: Artifacts
+          path: build/**/*
```

Comment:
I think you can narrow this down to just uv_run_tests_a.exe? That would probably help staying below that 500 MB limit (as would lowering retention to 1 or 2 days.)